src/cpp/rtps/history/History.cpp

```diff
@@ -100,6 +100,76 @@ void History::do_release_cache(
     change_pool_->release_cache(ch);
 }
 
+History::const_iterator History::find_change(
+        CacheChange_t* ch)
+{
+    if (mp_mutex == nullptr)
+    {
+        logError(RTPS_HISTORY, "You need to create a RTPS Entity with this History before using it");
+        return const_iterator();
+    }
+
+    std::lock_guard<RecursiveTimedMutex> guard(*mp_mutex);
+
+    return std::find_if(changesBegin(), changesEnd(), [this, ch](const CacheChange_t* chi)
+                   {
+                       // use the derived classes comparisson criteria for searching
+                       return this->matches_change(chi, ch);
+                   });
+}
+
+bool History::matches_change(
+        const CacheChange_t* ch_inner,
+        CacheChange_t* ch_outer)
+{
+    return ch_inner->sequenceNumber == ch_outer->sequenceNumber;
+}
+
+History::iterator History::remove_change(
+        const_iterator removal,
+        bool release)
```

Comment:
This declaration faces a use case that we found in the implementation of the new discovery server. Basically, History exposes iterators but doesn't allow to delete elements using them. In order to delete a new search was done within the old `remove_change` method. The actual code required to loop and delete was bulky, cumbersome and ugly as hell. We decided to add the new signature because:
- It follows STL collection philosophy.
- The old one can rely in the new one without any performance penalty.